middle_end/flambda2/to_cmm/to_cmm_primitive.ml

```diff
@@ -184,14 +184,26 @@ let make_array ~dbg kind alloc_mode args =
   | Naked_int64s -> C.allocate_unboxed_int64_array ~elements:args mode dbg
   | Naked_nativeints ->
     C.allocate_unboxed_nativeint_array ~elements:args mode dbg
+  | Unboxed_product _ ->
+    if P.Array_kind.must_be_gc_scannable kind
+    then C.make_alloc ~mode dbg ~tag:0 args
+    else
+      let mem_chunks =
+        P.Array_kind.element_kinds kind |> List.map C.memory_chunk_of_kind
```

Comment:
I don't understand this part of the code: shouldn't the list of elements kinds (which is iiuc the list of kinds of the unarized version of **each** element) somehow be duplicated by the number of products in the array being created ?

Reply:
Ah yes, this is wrong - fixing.